File: springboot-hsbc/src/main/java/com/hsbc/demo/service/EmployeeService.java
package com.hsbc.demo.service;

import java.util.List;

import com.hsbc.demo.exception.GenericException;
import com.hsbc.demo.model.Employee;
import com.hsbc.demo.model.Salary;

public interface EmployeeService {
	Salary getEmployeeSalary(long empId) throws GenericException;

	List<Employee> getEmployees();

	Employee getEmployee(long empId) throws GenericException;
}

File: springboot-hsbc/src/main/resources/static/js/app/EmployeeService.js
'use strict'
angular.module('demo.services', []).factory(
		'EmployeeService',
		[
				"$http",
				"CONSTANTS",
				function($http, CONSTANTS) {
					var service = {};
					service.getEmployeeById = function(empId) {
						var url = CONSTANTS.getAllEmployees + "/" + empId;
						return $http.get(url);
					}
					service.getAllEmployees = function() {
						return $http.get(CONSTANTS.getAllEmployees);
					}
					service.getEmployeeSalary = function(empId) {
						return $http.get(CONSTANTS.getAllEmployees + "/"
								+ empId + "/salary");
					}
					return service;
				} ]);